osu.Framework/Localisation/LocalisableStringDescriptor.cs

```diff
@@ -0,0 +1,151 @@
+        /// </summary>
+        /// <param name="key">The key to look for localisation by <see cref="LocalisationManager"/>.</param>
+        /// <param name="fallback">The fallback text to use when localised text is not found.</param>
+        /// <param name="args">The arguments to format the text with.</param>
+        /// <returns>The descriptor of the localisable string.</returns>
+        public static LocalisableStringDescriptor FromTranslatable(string key, string fallback, params object[] args)
+            => new LocalisableStringDescriptor(StringLocalisationType.Translation, fallback, key, args);
+
+        /// <summary>
+        /// Creates a new <see cref="LocalisableStringDescriptor"/>. This localises based on the value of <see cref="FrameworkSetting.Locale"/>.
+        /// Supposed usage:
+        /// <code>
+        /// LocalisableStringDescriptor.FromInterpolatedTranslatable($"You have played {count:N0} times!", "Played_count_self");
+        /// </code>
+        /// </summary>
+        /// <param name="interpolated">The interpolation string containing fallback and arguments.</param>
+        /// <param name="key">The key to look for localisation by <see cref="LocalisationManager"/>.</param>
+        /// <returns>The descriptor of the localisable string.</returns>
+        public static LocalisableStringDescriptor FromInterpolatedTranslatable(FormattableString interpolated, string key)
+            => new LocalisableStringDescriptor(StringLocalisationType.Translation, interpolated.Format, key, interpolated.GetArguments());
+
+        public bool TryGetTranslatable([NotNullWhen(true)] out string? key, [NotNullWhen(true)] out string? fallback, [NotNullWhen(true)] out object?[]? args)
+        {
+            if (type == StringLocalisationType.Translation)
+            {
+                key = string2;
+                fallback = string1;
+                args = this.args;
+                return true;
+            }
+            else
+            {
+                key = null;
+                fallback = null;
+                args = null;
+                return false;
+            }
+        }
+
+        private LocalisableStringDescriptor(StringLocalisationType type, string string1, string? string2, object?[]? args)
+        {
+            this.type = type;
+            this.string1 = string1;
+            this.string2 = string2;
+            this.args = args;
+        }
+
+        /// <summary>
+        /// Represents the type of a <see cref="LocalisableStringDescriptor"/>.
+        /// </summary>
+        private enum StringLocalisationType
+        {
+            /// <summary>
+            /// The <see cref="LocalisableStringDescriptor"/> is constructed from plain text and not localisable.
+            /// </summary>
+            None,
+
+            /// <summary>
+            /// The <see cref="LocalisableStringDescriptor"/> is constructed from Romanization/Unicode pair.
+            /// </summary>
+            Romanization,
```

Comment:
Annoying nitpick: should be
```suggestion
            Romanisation,
```
for consistency (American vs British spelling). Goes for every other place too.